remo/reports/api/views.py

```diff
@@ -19,3 +30,76 @@ def retrieve(self, request, pk):
         serializer = ActivitiesDetailedSerializer(report,
                                                   context={'request': request})
         return Response(serializer.data)
+
+
+class ActivitiesKPIFilter(django_filters.FilterSet):
+    """Filter for activities KPI endpoint."""
+    category = django_filters.CharFilter(name='functional_areas__name')
+    initiative = django_filters.CharFilter(name='campaign__name')
+
+    class Meta:
+        model = NGReport
+        fields = ['country', 'category', 'initiative']
+
+
+class ActivitiesKPIView(APIView):
+
+    def get(self, request):
+        """Returns serialized data for Activities KPI"""
+
+        qs = NGReport.objects.filter(report_date__lte=now())
+        activities = ActivitiesKPIFilter(request.query_params, queryset=qs)
+        weeks = int(request.query_params.get('weeks', KPI_WEEKS))
+
+        # Total calculations
+        total = activities.qs.count()
+
+        # Quarter calculations
+        current_quarter_start = get_quarter()[1]
+        quarter_total = activities.qs.filter(
+            report_date__gte=current_quarter_start).count()
+
+        # Total activities until start of quarter
+        quarter_s_total = activities.qs.filter(
+            report_date__lte=current_quarter_start).count()
+
+        try:
+            percent_quarter = (total-quarter_s_total)/float(quarter_s_total)
+        except ZeroDivisionError:
+            percent_quarter = 0
+
+        # Week calculations
```

Comment:
Please add comments to describe what you are doing here, because it's not clear without the specs to understand what's going on  - this applies to the events too.